### PR TITLE
Remove gcloud components update from e2e.sh

### PR DESCRIPTION
**Pull request by spxtr**
~~Please do not merge until #18863 is merged and running on both e2e Jenkins and PR Jenkins.~~ Done. Ready for merge.
#18846

### CODE REVIEW

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -1177,45 +1178,6 @@ if [[ "${E2E_UP,,}" == "true" || "${JENKINS_FORCE_GET_TARS:-}" =~ ^[yY]$ ]]; the
             exit 1
         fi
 
-        # Tell kube-up.sh to skip the update, it doesn't lock. An internal
-        # gcloud bug can cause racing component updates to stomp on each
-        # other.
-        export KUBE_SKIP_UPDATE=y
```
Comment (jlowdermilk) [inline hack/jenkins/e2e.sh]: This one you should keep, at least until we remove the `gcloud components install` calls from [cluster/gce/util.sh](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh#L82)

Comment (spxtr) [inline hack/jenkins/e2e.sh]: I added it in on line 1142 so that we never update in an e2e test.

**Comment by spxtr** [comment]
@ixdy I had an error, it was saying `export GCE_SERVICE_ACCOUNT:=$(...)`. The extra colon got caught by PR Jenkins since it builds and then tests using the version of e2e.sh in the PR.

**Comment by spxtr** [comment]
@k8s-bot unit test this

**Comment by spxtr** [comment]
Manually merging to watch Jenkins.

**Pull request merged**